src/context/auth.context.tsx

```diff
@@ -0,0 +1,83 @@
+import { createContext, ReactNode, useEffect, useState } from "react";
+import { Customer } from "../interfaces/customers/customer.interface";
+import { useFetchCustomer } from "../hooks/customers/useFetchCustomer";
+
+export interface IAuthContext {
+  customer: Customer | null;
+  accessToken: string | null;
+  refreshToken: string | null;
+  login: (
```

Comment:
You can create a general method of setting the things for login and use it is useLogin hook. In this case you don't have to fetch the user here. You fetch it there in the login method.